src/main/java/org/stanford/ncbo/owlapi/wrapper/OntologyParser.java

```diff
@@ -162,21 +184,22 @@ private void addGroundMetadata(IRI documentIRI, OWLDataFactory factory, OWLOntol
 			targetOwlManager.addAxiom(targetOwlOntology, annotationAssertionAxiom);
 
 			if (isFile && (annotationProperty.toString().contains("versionInfo"))) {
-				IRI versionSubjectIRI = IRI.create("http://bioportal.bioontology.org/ontologies/versionSubject");
+				IRI versionSubjectIRI = IRI.create(VERSION_SUBJECT);
 				OWLAnnotationProperty versionAnnotationProperty = factory.getOWLAnnotationProperty(OWLRDFVocabulary.OWL_VERSION_INFO.getIRI());
 				OWLAnnotationAssertionAxiom versionAnnotationAssertionAxiom = factory.getOWLAnnotationAssertionAxiom(versionAnnotationProperty, versionSubjectIRI, annotationValue);
 				targetOwlManager.addAxiom(targetOwlOntology, versionAnnotationAssertionAxiom);
 			}
 		}
 	}
 
-	private boolean buildOWLOntology(boolean isOBO) {
+	private boolean buildOWLOntology(OWLOntology masterOntology, boolean isOBO) {
```

Comment:
I think the desired functionality of setting the target ontology IRI can be accomplished with a lot less code. This strikes me as something that fits naturally into the `addGroundMetadata` method, which loosely speaking initializes the metadata for the target ontology.

Instead of modifying the method signature for `buildOWLOntology` and adding a new method (`addOntologyIRI`), you could simply add an else clause on [line 154](https://github.com/ncbo/owlapi_wrapper/blob/b70dbcb1cfe2d5207a57b3e2b6192b6573af30e0/src/main/java/org/stanford/ncbo/owlapi/wrapper/OntologyParser.java#L154) in `addGroundMetadata`, e.g.:

```java
if (ontologyID.isAnonymous()) {
  return;
} else {
  OWLOntologyID owlOntologyID = new OWLOntologyID(ontologyID.getOntologyIRI(), ontologyID.getVersionIRI());
  SetOntologyID setOntologyID = new SetOntologyID(targetOwlOntology, owlOntologyID);
  targetOwlManager.applyChange(setOntologyID);
}
```

Reply:
`addGroundMetadata` method, do a  loop for each ontology in the `sourceOwlManager` (the master ontology + imported ontologies). I need to save only the master ontology ontology IRI.

The else clause of the `ontologyID.isAnonymous()` condition will be all the ontologies that do not have an IRI, wich means that  `ontologyID.getOntologyIRI()` will be **null** (you can see the definition of the `isAnonymous()` method bellow), so we can't use is to set the ontology IRI.

 ```java
  public boolean isAnonymous() {
        return !this.ontologyIRI.isPresent();
    }
```

(I hope that I was clear)

Reply:
Ugh. I forgot about the imports. :tired_face: